rtos/TARGET_CORTEX/SysTimer.cpp

```diff
@@ -106,16 +114,19 @@ uint32_t SysTimer::resume()
     _suspend_time_passed = true;
     remove();
 
-    uint64_t new_tick = (ticker_read_us(_ticker_data) - _start_time) * OS_TICK_FREQ / 1000000;
-    if (new_tick > _tick) {
+    const timestamp_t elpased_us = ticker_read_us(_ticker_data) - _time_us;
```

Comment:
`elpased`->`elapsed`

Is it worth sticking an explicit cast in for the narrowing here?

It's probably worth sticking a comment here that the extra complexity is being employed to avoid the 64-bit division in a critical section, lest anyone touch it in future.